test/e2e/node/pod_resize.go

```diff
@@ -0,0 +1,1421 @@
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			pod, err := podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			differs := false
+			for idx, c := range pod.Spec.Containers {
+				if diff.ObjectDiff(c.Resources, *pod.Status.ContainerStatuses[idx].Resources) != "" {
+					differs = true
+					break
+				}
+			}
+			if differs {
+				continue
+			}
+			return pod, nil
+		}
+		return nil, fmt.Errorf("timed out waiting for pod spec resources to match pod status resources")
+	}
+	rsErr := waitForContainerRestart()
+	framework.ExpectNoError(rsErr, "failed waiting for expected container restart")
+	// Wait for pod resource allocations to equal expected values after resize
+	resizedPod, aErr := waitPodAllocationsEqualsExpected()
+	framework.ExpectNoError(aErr, "failed to verify pod resource allocation values equals expected values")
+	// Wait for container cgroup values to equal expected cgroup values after resize
+	cErr := waitContainerCgroupValuesEqualsExpected()
+	framework.ExpectNoError(cErr, "failed to verify container cgroup values equals expected values")
+	//TODO(vinaykul,InPlacePodVerticalScaling): Remove featureGatePostAlpha upon exiting Alpha.
+	//                containerd needs to add CRI support before Beta (See Node KEP #2273)
+	if featureGatePostAlpha {
+		// Wait for PodSpec container resources to equal PodStatus container resources indicating resize is complete
+		rPod, rErr := waitPodStatusResourcesEqualSpecResources()
+		framework.ExpectNoError(rErr, "failed to verify pod spec resources equals pod status resources")
+
+		ginkgo.By("verifying pod status after resize")
+		verifyPodStatusResources(rPod, expectedContainers)
+	}
+	return resizedPod
+}
+
+func doPodResizeTests() {
+	f := framework.NewDefaultFramework("pod-resize")
+
+	ginkgo.BeforeEach(func() {
+		podClient = f.PodClient()
+		ns = f.Namespace.Name
+	})
+
+	type testCase struct {
+		name        string
+		containers  []TestContainerInfo
+		patchString string
+		expected    []TestContainerInfo
+	}
+
+	noRestart := v1.RestartNotRequired
+	doRestart := v1.RestartRequired
+	tests := []testCase{
+		{
+			name: "Guaranteed QoS pod, one container - increase CPU & memory",
+			containers: []TestContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &ContainerResources{CPUReq: "100m", CPULim: "100m", MemReq: "200Mi", MemLim: "200Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &noRestart,
+				},
+			},
+			patchString: `{"spec":{"containers":[
+						{"name":"c1", "resources":{"requests":{"cpu":"200m","memory":"400Mi"},"limits":{"cpu":"200m","memory":"400Mi"}}}
+					]}}`,
+			expected: []TestContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &ContainerResources{CPUReq: "200m", CPULim: "200m", MemReq: "400Mi", MemLim: "400Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &noRestart,
+				},
+			},
```

Comment:
resizeToOriginalPatchString value is 

resizeToOriginalPatchString: `{"spec":{"containers":[
				{"name":"c1", "resources":{"requests":{"cpu":"100m","memory":"200Mi"},"limits":{"cpu":"100m","memory":"200Mi"}}}
			]}}`,